testsuite fails with ghc-9.10.2

For Stackage Nightly

```
       Test suite failure for package ghc-typelits-natnormalise-0.7.10                                          17:33:23 [58/982445]
           unit-tests:  exited with: ExitFailure 1

:

                 (a <=? a+1) ~ False:                                                                                             FA
IL                                                                                                                                  
                   tests/Tests.hs:652:                                                                                              
                   tests/ErrorTests.hs:191:15: error: [GHC-18872]                                                                   
                       • Couldn't match type ‘ghc-internal-9.1002.0:GHC.Internal.Data.Type.Ord.OrdCond                              
                                                (CmpNat a (a + 2)) True True False’                                                 
                                        with ‘False’                                                                                
                           arising from a use of ‘proxyInEq'’                                                                       
                       • In the expression: proxyInEq'            
                         In an equation for ‘testProxy10’: testProxy10 = proxyInEq'                                                 
                       • Relevant bindings include                                                                                  
                           testProxy10 :: Proxy a -> Proxy (a + 2) -> ()                                                            
                             (bound at tests/ErrorTests.hs:191:1)                                                                   
                   (deferred type error)                                                                                            
                   Use -p '/(a <=? a+1) ~ False/' to rerun this test only.
:
                 2a <=? 4a ~ False:                                                                                               FA
IL                                                                                                                                  
                   tests/Tests.hs:652:                                                                                              
                   tests/ErrorTests.hs:334:15: error: [GHC-18872]                                                                   
                       • Couldn't match type ‘ghc-internal-9.1002.0:GHC.Internal.Data.Type.Ord.OrdCond                              
                                                (CmpNat (2 * a) (4 * a)) True True False’                                           
                                        with ‘False’                                                                                
                           arising from a use of ‘proxyInEq'’                                                                       
                       • In the expression: proxyInEq'                                                                              
                         In an equation for ‘testProxy14’: testProxy14 = proxyInEq'
                       • Relevant bindings include
                           testProxy14 :: Proxy (2 * a) -> Proxy (4 * a) -> ()
                             (bound at tests/ErrorTests.hs:334:1)
                   (deferred type error)
                   Use -p '/2a <=? 4a ~ False/' to rerun this test only.
```

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -33,6 +33,11 @@ jobs:
             compilerVersion: 9.12.1
             setup-method: ghcup
             allow-failure: false
+          - compiler: ghc-9.10.2
+            compilerKind: ghc
+            compilerVersion: 9.10.2
+            setup-method: ghcup
+            allow-failure: false
           - compiler: ghc-9.10.1
             compilerKind: ghc
             compilerVersion: 9.10.1

--- a/tests/ErrorTests.hs
+++ b/tests/ErrorTests.hs
@@ -28,6 +28,10 @@ import GHC.IO.Encoding            (getLocaleEncoding, textEncodingName, utf8)
 import Language.Haskell.TH        (litE, stringL)
 import Language.Haskell.TH.Syntax (runIO)
 
+#if __GLASGOW_HASKELL__ >= 901
+import qualified Data.Type.Ord
+#endif
+
 testProxy1 :: Proxy (x + 1) -> Proxy (2 + x)
 testProxy1 = id
 
@@ -194,8 +198,8 @@ testProxy10Errors =
 #if __GLASGOW_HASKELL__ >= 912
   [$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8
-          then litE $ stringL "Couldn't match type ‘ghc-internal-9.1201.0:GHC.Internal.Data.Type.Ord.OrdCond"
-          else litE $ stringL "Couldn't match type `ghc-internal-9.1201.0:GHC.Internal.Data.Type.Ord.OrdCond"
+          then litE $ stringL "Couldn't match type ‘Data.Type.Ord.OrdCond"
+          else litE $ stringL "Couldn't match type `Data.Type.Ord.OrdCond"
     )
   ,$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8
@@ -211,8 +215,8 @@ testProxy10Errors =
 #elif __GLASGOW_HASKELL__ >= 910
   [$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8
-          then litE $ stringL "Couldn't match type ‘ghc-internal-9.1001.0:GHC.Internal.Data.Type.Ord.OrdCond"
-          else litE $ stringL "Couldn't match type `ghc-internal-9.1001.0:GHC.Internal.Data.Type.Ord.OrdCond"
+          then litE $ stringL "Couldn't match type ‘Data.Type.Ord.OrdCond"
+          else litE $ stringL "Couldn't match type `Data.Type.Ord.OrdCond"
     )
   ,$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8
@@ -354,8 +358,8 @@ testProxy14Errors =
 #if __GLASGOW_HASKELL__ >= 912
   [$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8
-          then litE $ stringL "Couldn't match type ‘ghc-internal-9.1201.0:GHC.Internal.Data.Type.Ord.OrdCond"
-          else litE $ stringL "Couldn't match type `ghc-internal-9.1201.0:GHC.Internal.Data.Type.Ord.OrdCond"
+          then litE $ stringL "Couldn't match type ‘Data.Type.Ord.OrdCond"
+          else litE $ stringL "Couldn't match type `Data.Type.Ord.OrdCond"
     )
   ,$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8
@@ -371,8 +375,8 @@ testProxy14Errors =
 #elif __GLASGOW_HASKELL__ >= 910
   [$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8
-          then litE $ stringL "Couldn't match type ‘ghc-internal-9.1001.0:GHC.Internal.Data.Type.Ord.OrdCond"
-          else litE $ stringL "Couldn't match type `ghc-internal-9.1001.0:GHC.Internal.Data.Type.Ord.OrdCond"
+          then litE $ stringL "Couldn't match type ‘Data.Type.Ord.OrdCond"
+          else litE $ stringL "Couldn't match type `Data.Type.Ord.OrdCond"
     )
   ,$(do localeEncoding <- runIO (getLocaleEncoding)
         if textEncodingName localeEncoding == textEncodingName utf8

